contracts/pallet-ibc/src/client.rs

```diff
@@ -118,55 +118,45 @@ where
 		client_id: &ClientId,
 		height: Height,
 	) -> Result<Option<AnyConsensusState>, ICS02Error> {
-		let mut cs_states = ConsensusStates::<T>::iter_key_prefix(client_id)
-			.map(|(height, cs_state)| {
-				let cs = AnyConsensusState::decode_vec(&cs_state).map_err(|e| {
-					ICS02Error::implementation_specific(format!(
-						"[next_consensus_state]: error decoding consensus state from bytes {}",
-						e
-					))
-				})?;
-				Ok((height, cs))
+		let consensus_heights = ConsensusHeights::<T>::get(client_id.as_bytes().to_vec());
+		let cs_state = consensus_heights
+			.into_iter()
+			.filter(|(revision_number, revision_height)| {
+				let next_height = Height::new(*revision_number, *revision_height);
+				next_height > height
 			})
-			.collect::<Result<Vec<_>, ICS02Error>>()?;
-
-		cs_states.sort_by(|a, b| a.0.cmp(&b.0));
-
-		for cs in cs_states {
-			if cs.0 > height {
-				return Ok(Some(cs.1))
-			}
-		}
+			.take(1)
+			.next()
```

Comment:
is this `next relevant`? If the iterator has only one element, and you're mapping over it, would not that be enough?